external-crates/move/move-vm/runtime/src/plugin.rs

```diff
@@ -0,0 +1,64 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use std::sync::Arc;
+
+use move_binary_format::{
+    errors::{PartialVMResult, VMResult},
+    file_format::Bytecode,
+};
+use move_core_types::account_address::AccountAddress;
+use move_vm_types::{loaded_data::runtime_types::Type, values::Locals};
+
+use crate::{
+    interpreter::{FrameInterface, InstrRet, InterpreterInterface},
+    loader::{Function, Loader, Resolver},
+};
+
+pub(crate) trait Plugin {
+    fn pre_hook_entrypoint(
```

Comment:
This seems like an init/constructor. We should probably have a de-init/destructor too for plugins which need to cleanup.